src/backend/SwapChain.cpp

```
@@ -0,0 +1,98 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "backend/SwapChain.h"
+
+#include "backend/Device.h"
+#include "backend/Texture.h"
+
+namespace backend {
+
+    // SwapChain
+
+    SwapChainBase::SwapChainBase(SwapChainBuilder* builder)
+            : device(builder->device), implementation(builder->implementation) {
+    }
+
+    SwapChainBase::~SwapChainBase() {
+    }
+
+    DeviceBase* SwapChainBase::GetDevice() {
+        return device;
+    }
+
+    void SwapChainBase::Configure(nxt::TextureFormat format, uint32_t width, uint32_t height) {
+        this->format = format;
+        this->width = width;
+        this->height = height;
+    }
+
+    TextureBase* SwapChainBase::GetNextTexture() {
+        auto* builder = device->CreateTextureBuilder();
```

We should return an error if Configure wasn't called before.